front/production_checks/checks/nango_connection_id_cleanup_slack.ts

```diff
@@ -0,0 +1,81 @@
+import { Nango } from "@nangohq/node";
+import { QueryTypes, Sequelize } from "sequelize";
+
+import { CheckFunction } from "@app/production_checks/types/check";
+
+const {
+  CONNECTORS_DATABASE_READ_REPLICA_URI,
+  NANGO_SECRET_KEY,
+  NANGO_SLACK_CONNECTOR_ID,
+} = process.env;
+
+export const nangoConnectionIdCleanupSlack: CheckFunction = async (
+  checkName,
+  reportSuccess,
+  reportFailure
+) => {
+  if (!NANGO_SECRET_KEY) {
+    throw new Error("Env var NANGO_SECRET_KEY is not defined");
+  }
+  if (!NANGO_SLACK_CONNECTOR_ID) {
+    throw new Error("Env var NANGO_SLACK_CONNECTOR_ID is not defined");
+  }
+  if (!CONNECTORS_DATABASE_READ_REPLICA_URI) {
+    throw new Error(
+      "Env var CONNECTORS_DATABASE_READ_REPLICA_URI is not defined"
+    );
+  }
+
+  // Get all the Slack configurations in the database
+  const connectorsSequelize = new Sequelize(
+    CONNECTORS_DATABASE_READ_REPLICA_URI,
+    {
+      logging: false,
+    }
+  );
+  const dbSlackConfigurations: { id: number; slackTeamId: string }[] =
```

Comment:
Can we make `dbSlackConfigurations` a `Set` of SlackTeamId instead of an array because the look up time done at line 65 with `if (!dbSlackConfigurations.find((sc) => sc.slackTeamId === slackTeamId)) {` is going to grow slower quadratically.

Reply:
Done!